spark/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -1607,6 +1607,20 @@ class DeltaSuite extends QueryTest
     }
   }
 
+  test("support Java8 time objects") {
+    withSQLConf(SQLConf.DATETIME_JAVA8API_ENABLED.key -> "true") {
+      val tableName = "my_table"
+      withTable(tableName) {
+        spark.sql(s"CREATE TABLE $tableName (id STRING, date DATE) USING DELTA;")
```

Comment:
Add a column with `TIMESTAMP` type to test LocalDateTime.
Please make sure that the issue repro with `TIMESTAMP` as well.

Reply:
Good catch, tested with `TIMESTAMP` which uses `time.Instant` not `time.LocalDateTime`

Reply:
Also verified that the test does repro for timestamp.